File: LeetCode/ProblemSet/Volume (1200-1299)/LeetCode1260/LeetCode1260-Solution-01.cpp
#include <bits/stdc++.h>

class Solution {
public:
    std::vector<std::vector<int>> shiftGrid(std::vector<std::vector<int>>& grid, int k) {
        const int n = (int) grid.size();
        const int m = (int) grid[0].size();

        int total = n * m;
        std::vector<std::vector<int>> ans(n, std::vector<int>(m, 0));
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                int id = (i * m + j + k) % total;
                ans[id / m][id % m] = grid[i][j];
            }
        }

        return ans;
    }
};

File: LeetCode/ProblemSet/Volume (1200-1299)/LeetCode1260/LeetCode1260-Solution-01.py
from typing import *

class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        n, m = len(grid), len(grid[0])

        total = n * m
        ans = [[0] * m for _ in range(n)]
        for i, row in enumerate(grid):
            for j, x in enumerate(row):
                id = (i * m + j + k) % total
                ans[id // m][id % m] = x
        
        return ans